pyinfra/api/connectors/ansible.py

```diff
@@ -133,23 +143,46 @@ def _parse_ansible_hosts(hosts):
             yield host
 
 
-def parse_inventory_tree(inventory_tree, host_to_groups=dict(), group_stack=set()):
+def parse_inventory_tree(inventory_tree, host_to_groups=dict(), group_stack=set(), group_varz=dict()):
     for group in inventory_tree:
         # set logic adds tolerance for duplicate group names
         groups = group_stack.union({group})
 
+        if 'vars' in inventory_tree[group]:
+            group_varz[group] = inventory_tree[group]['vars']
+
         if 'hosts' in inventory_tree[group]:
             for host in inventory_tree[group]['hosts']:
-                append_groups_to_host(host, groups, host_to_groups)
+                append_groups_to_host(host, groups, host_to_groups, group_varz)
 
         if 'children' in inventory_tree[group]:
             # recursively parse inventory tree
-            parse_inventory_tree(inventory_tree[group]['children'], host_to_groups, groups)
+            parse_inventory_tree(inventory_tree[group]['children'], host_to_groups, groups, group_varz)
 
     return host_to_groups
 
 
-def append_groups_to_host(host, groups, host_to_groups):
+
+def parse_inventory_tree_vars(inventory_tree, host_to_groups=dict(), group_stack=set(), group_varz=dict()):
```

Comment:
I'm not 100% sure this function is needed? I think the `parse_inventory_tree` could also handle this (returning group vars as well as host to groups).